### ISSUE TITLE
Unhandled exception: Crash on update

### ISSUE THREAD

**Issue by BubbaWPro** [remyroy/CDDA-Game-Launcher#19]
Description: Began updating, after the files appeared to have finished downloading, a few seconds pass, and the program crashes. This was the first time I had used the launcher since 0.3. Assuming you see this, keep up the good work bro. I'm running Windows 7 64.
Version: 0.6.5
Type: `<class 'OSError'>`
Value: [WinError 145] The directory is not empty: 'C:\\Users\\ADMIN\\Desktop\\CDDA\\previous_version\\lang\\mo\\it_IT'
Traceback:
```
  File "C:\Users\Remy\Projects\CDDA-Game-Launcher\cddagl\ui.py", line 1435, in download_http_finished
  File "C:\Users\Remy\Projects\CDDA-Game-Launcher\cddagl\ui.py", line 1451, in backup_current_game
  File "c:\python34\lib\shutil.py", line 482, in rmtree
  File "c:\python34\lib\shutil.py", line 372, in _rmtree_unsafe
  File "c:\python34\lib\shutil.py", line 372, in _rmtree_unsafe
  File "c:\python34\lib\shutil.py", line 372, in _rmtree_unsafe
  File "c:\python34\lib\shutil.py", line 381, in _rmtree_unsafe
  File "c:\python34\lib\shutil.py", line 379, in _rmtree_unsafe

```

**Comment by remyroy** [comment]
I don't think there is much we can do for this type of error. It seems we cannot remove that directory because it it being used by another process. I think the solution will be to try to find out which process it is and ask the user to retry the operation or cancel the update.

I'll try to add the message box soon.

**Comment by remyroy** [comment]
The use of [the psutil package](https://pypi.python.org/pypi/psutil) could be useful here. I'm leaving a note for myself.

**Comment by remyroy** [comment]
This is fixed with 3f2d6b38fa262b08c2d2b6911785f61667f9efa8. There will now be a retry message box shown if we cannot remove the `previous_version` directory. That message box will show if a specific process is holding a handle on a contentious file or directory.

This fix will be part of the next release.
